File: SBZ-CDSS-Frontend/src/app/services/administrator/administrator.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs/Observable';
import { Lekar } from '../../model/Lekar';
import { Pacijent } from '../../model/Pacijent';
import { Simptom } from '../../model/Simptom';
import { Bolest } from '../../model/Bolest';
import { Lek } from '../../model/Lek';

@Injectable()
export class AdministratorService {

    constructor(
        private http: HttpClient
    ) { }

    dodajLekara(lekar: Object): Observable<Lekar> {
        return this.http.post<Lekar>("/api/lekar/dodaj", lekar);
    }

    lekar(lekarId: number): Observable<Lekar> {
        return this.http.get<Lekar>("api/lekar/" + lekarId);
    }

    lekari(page, size, filter) {
        if (!filter) {
            const params:HttpParams = new HttpParams().set("page", page).append("size", size);
            return this.http.get("/api/lekari", {params, observe: "response"});
        } else {
            const params:HttpParams = new HttpParams().set("page", page).append("size", size).append("filter", filter);
            return this.http.get("/api/lekari", {params, observe: "response"});
        }
    }

    dodajPacijenta(pacijent: Object): Observable<Pacijent> {
        return this.http.post<Pacijent>("/api/pacijent/dodaj", pacijent);
    }

    pacijent(pacijentId: number): Observable<Pacijent> {
        return this.http.get<Pacijent>("api/pacijent/" + pacijentId);
    }

    pacijenti(page, size, filter) {
        if (!filter) {
            const params:HttpParams = new HttpParams().set("page", page).append("size", size);
            return this.http.get("/api/pacijenti", {params, observe: "response"});
        } else {
            const params:HttpParams = new HttpParams().set("page", page).append("size", size).append("filter", filter);
            return this.http.get("/api/pacijenti", {params, observe: "response"});
        }
    }

    dodajSimptom(simptom: Object): Observable<Simptom> {
        return this.http.post<Simptom>("/api/simptom/dodaj", simptom);
    }

    simptomi(page, size, filter) {
        if (!filter) {
            const params:HttpParams = new HttpParams().set("page", page).append("size", size);
            return this.http.get("/api/simptomi", {params, observe: "response"});
        } else {
            const params:HttpParams = new HttpParams().set("page", page).append("size", size).append("filter", filter);
            return this.http.get("/api/simptomi", {params, observe: "response"});
        }
    }

    dodajBolest(bolest: Object): Observable<Bolest> {
        return this.http.post<Bolest>("/api/bolest/dodaj", bolest);
    }

    bolest(bolestId: number): Observable<Bolest> {
        return this.http.get<Bolest>("api/bolest/" + bolestId);
    }

    bolesti(page, size, filter) {
        if (!filter) {
            const params:HttpParams = new HttpParams().set("page", page).append("size", size);
            return this.http.get("/api/bolesti", {params, observe: "response"});
        } else {
            const params:HttpParams = new HttpParams().set("page", page).append("size", size).append("filter", filter);
            return this.http.get("/api/bolesti", {params, observe: "response"});
        }
    }

    dodajLek(lek: Object): Observable<Lek> {
        return this.http.post<Lek>("/api/lek/dodaj", lek);
    }

    lek(lekId: number): Observable<Lek> {
        return this.http.get<Lek>("api/lek/" + lekId);
    }

    lekovi(page, size, filter) {
        if (!filter) {
            const params:HttpParams = new HttpParams().set("page", page).append("size", size);
            return this.http.get("/api/lekovi", {params, observe: "response"});
        } else {
            const params:HttpParams = new HttpParams().set("page", page).append("size", size).append("filter", filter);
            return this.http.get("/api/lekovi", {params, observe: "response"});
        }
    }

}

File: SBZ-CDSS-App/src/main/java/sbz/cdss/services/AdministratorService.java
package sbz.cdss.services;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import sbz.cdss.facts.Bolest;
import sbz.cdss.facts.Lek;
import sbz.cdss.facts.Lekar;
import sbz.cdss.facts.Pacijent;
import sbz.cdss.facts.Simptom;

public interface AdministratorService {
	
	Lekar dodajLekara(Lekar lekar);
	Lekar lekar(Long id);
	Page<Lekar> lekari(Pageable pageable, String filter);
	
	Pacijent dodajPacijenta(Pacijent pacijent);
	Pacijent pacijent(Long id);
	Page<Pacijent> pacijenti(Pageable pageable, String filter);
	Pacijent izmeniPacijenta(Long id, Pacijent pacijent);
	
	Simptom dodajSimptom(Simptom simptom);
	Page<Simptom> simptomi(Pageable pageable, String filter);
	void obrisiSimptom(Long id);
	
	Bolest dodajBolest(Bolest bolest);
	Bolest bolest(Long id);
	Page<Bolest> bolesti(Pageable pageable, String filter);
	Bolest izmeniBolest(Long id, Bolest bolest);
	void obrisiBolest(Long id);
	
	Lek dodajLek(Lek lek);
	Lek lek(Long id);
	Page<Lek> lekovi(Pageable pageable, String filter);
	Lek izmeniLek(Long id, Lek lek);
	void obrisiLek(Long id);

}
